### PR TITLE
Basic popover added with Start/Stop Injecting option

### PR DESCRIPTION
**Pull request by sammyshj**
Basic popover has been added with Start/Stop Injecting option.
The images and the css has been copied from the privly-chrome repo.
Presently, when the `Start/Stop Injecting` option is clicked in the popover, the `modal_button.js` sends a message with `name: modeChange` and `message: on`(if the Start Injecting option is pressed) or `message: off`(if the Stop Injecting option is pressed).
In the `reading_process.js`, the `handleMessage()` function is implemented, which on receiving the `modeChange` message, calls `privly.start()` or `privly.stop()` accordingly.
@smcgregor Let me know if you think the changes are okay, then, I will add more comments to the files so that they can be merged.
All the tests pass and can be seen at https://travis-ci.org/sammyshj/privly-safari/builds/66632406

### CODE REVIEW

**Comment by smcgregor** [comment]
I think it would be better to reference the bootstrap found in the privly-applications so it isn't duplicated throughout the code base.

In general I think the implementation is looking good, but it would be better to have all the function level docs done before I conduct a thorough review.

**Comment by sammyshj** [comment]
@smcgregor Okay, I will refer to the bootstrap files in privly-applications repo.
Should I update the docs right now with the implementation of Start/Stop Injecting? Or should I wait for https://github.com/privly/privly-applications/pull/250 and leave the message sending and receiving for then?

**Comment by sammyshj** [comment]
@smcgregor I have removed the redundant bootstrap and jquery files. Let me know, what should be done about using the message sending/receiving as referred to in the above comment.
Thanks

**Comment by smcgregor** [comment]
In general you should always have the function level and file level comments written before you ask for input from others, otherwise it will be difficult to reconstruct what each function is supposed to be doing.

**Comment by sammyshj** [comment]
@smcgregor I have added the comments for all the files. Now on clicking `Start/Stop Injecting`, the `modalButton.modeChange()` is called from the popover. This then changes the `modalButton.badgeText` value and then calls the `readingProcess.tabsChange()`. Then, the modeChange message is dispatched from `readingProcess.tabChange()`. The incoming messages are finally handled in the `handle_privly_execution.js` file.
Let me know if the code looks good and I will go ahead and squash the commits.

**Comment by smcgregor** [comment]
I'll take a look in about 10 hours :)

File: privly.safariextension/scripts/modal_button.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * @namespace for the modal button.
+ */
+var modalButton = {
+
+  /**
+   * Gives the current state of the modal button.
+   */
+  badgeText: "on",
+
+
```
Comment (smcgregor) [inline privly.safariextension/scripts/modal_button.js]: nit: extra whitespace

File: privly.safariextension/scripts/popover.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * you click on the Privly icon in the browser Safari.
+ */
+
+/**
+ * Helper function for activateExtension() and deactivateExtension().
+ */
+function extensionStateChange(toShow, toHide) {
+
+  safari.extension.globalPage.contentWindow.modalButton.modeChange();
+  // Update the UI
```
Comment (smcgregor) [inline privly.safariextension/scripts/popover.js]: nit: should have blank line before the comment.

File: privly.safariextension/scripts/popover.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileOverview This file provides for the menu that pops over when
+ * you click on the Privly icon in the browser Safari.
+ */
+
+/**
+ * Helper function for activateExtension() and deactivateExtension().
+ */
```
Comment (smcgregor) [inline privly.safariextension/scripts/popover.js]: You need @params in the comments for any function with parameters.

**Comment by smcgregor** [comment]
I have just a few comments in addition to the inline comments above:

* The popover's cursor should change to a pointer when hovering over the button.
* The button's state is not currently respected when opening a new page, it always injects.
* You should organize your scripts into separate folders for content and background scripts.

**Comment by sammyshj** [comment]
All the points mentioned in the above comment have been implemented in https://github.com/sammyshj/privly-safari/commit/09b5fc7b132cb4c8169dc08f9200b6d0134e0aa8
Let me know if I should squash the commits.
Thanks

**Comment by sammyshj** [comment]
@smcgregor Squashed all commits into one. :)

**Comment by smcgregor** [comment]
Merging.

What are you working on now? Unless you are in the middle of something I think we should hook in coveralls like it is in [privly-applications](https://coveralls.io/builds/2689611).

**Pull request merged**